src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs

```diff
@@ -16,7 +16,12 @@
 
 namespace Microsoft.CodeAnalysis.FindSymbols
 {
-    using RelatedTypeCache = ConditionalWeakTable<Solution, ConcurrentDictionary<(SymbolKey, IImmutableSet<Project>), AsyncLazy<ImmutableArray<(SymbolKey, ProjectId)>>>>;
+    // Must cache using SymbolKey+ProjectId.  That's because the same symbol key may be found among many projects, but
+    // the same operation on the same symbol key might produce different results depending on which project it was found
+    // in.  For example, each symbol's project may have a different set of downstream dependent projects.  As such,
+    // there may be a different set of related symbols found for each.
+    using RelatedTypeCache = ConditionalWeakTable<Solution, ConcurrentDictionary<(SymbolKey, ProjectId, IImmutableSet<Project>), AsyncLazy<ImmutableArray<(SymbolKey, ProjectId)>>>>;
```

Comment:
❔ Do we have any regression tests in place for this change (i.e. if we went back to not including `ProjectId` in the key, would tests start to fail)?

Reply:
yes.  the added test in this PR will fail that case.